### PR TITLE
EVG-14218 Add optional parameter to gotest.parse_files

### PR DESCRIPTION
**Pull request by bynn**
[EVG-14218](https://jira.mongodb.org/browse/EVG-14218)

### Description 
does not return error when len == 0

### Testing 
  < add a description of how you tested it >

### CODE REVIEW

File: agent/command/results_gotest.go [hunk 1]
```diff
@@ -22,6 +22,11 @@ type goTestResults struct {
 	// a list of filename blobs to include
 	// e.g. "monitor.suite", "output/*"
 	Files []string `mapstructure:"files" plugin:"expand"`
+
+	// Optional, when set to true, causes this command to be skipped over without an error when
+	// no files are found to be parsed.
+	Optional bool `mapstructure:"optional" plugin:"expand"`
```
Comment (ablack12) [inline agent/command/results_gotest.go:28]: I think Optional implies that all errors should be ignored, whereas here we just want to ignore empty output, so maybe OptionalOutput or etc would be more informative

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: agent/command/results_gotest.go (hunk 1 above)
Comment (bynn) [reply agent/command/results_gotest.go]: done

File: agent/command/results_gotest.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ type goTestResults struct {
 
 	// Optional, when set to true, causes this command to be skipped over without an error when
 	// no files are found to be parsed.
-	Optional bool `mapstructure:"optional" plugin:"expand"`
+	OptionalOutput bool `mapstructure:"optional" plugin:"expand"`
```
Comment (ablack12) [inline agent/command/results_gotest.go:28]: nit: mapstructure/comments should match

Comment (bynn) [reply agent/command/results_gotest.go]: done

**Review by ablack12 — APPROVED** [review]

**Comment by ablack12** [comment]
You'll want to add this parameter to the [wiki page](https://github.com/evergreen-ci/evergreen/wiki/Project-Commands) and probably link it when you close the ticket.

**Review by malikchaya2 — APPROVED** [review]

**Comment by bynn** [comment]
https://github.com/evergreen-ci/evergreen/wiki/Project-Commands#gotest-parse-files

Updated description

**Pull request merged**